katdal/applycal.py

```diff
@@ -87,6 +152,13 @@ def add_applycal_sensors(cache, attrs, freqs):
                      for (ant_idx, ant) in enumerate(cal_ants)}
     if not cal_input_map:
         return
+    try:
+        cal_spw = SpectralWindow(attrs['cal_center_freq'], None,
+                                 attrs['cal_n_chans'], sideband=1,
+                                 bandwidth=attrs['cal_bandwidth'])
+        cal_freqs = cal_spw.channel_freqs
+    except KeyError:
+        return
```

Comment:
Done.

Reply:
Picked option 1 for now.

Reply:
I'm happy with option 1, because the semantic is consistently "if I didn't find enough information to apply cal, proceed without it", whether that's because there is no cal information (cal_input_map is empty) or it is incomplete.